corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -176,14 +176,15 @@ hqDefine("cloudcare/js/formplayer/menus/api", [
                 };
                 var casesPerPage = parseInt($.cookie("cases-per-page-limit"))
                     || (window.innerWidth <= constants.SMALL_SCREEN_WIDTH_PX ? 5 : 10);
+                var formplayerSessionPreferredLang = $.cookie("formplayer_session_lang");
                 const data = {
                     "username": user.username,
                     "restoreAs": user.restoreAs,
                     "domain": user.domain,
                     "app_id": appId,
                     "endpoint_id": params.endpointId,
                     "endpoint_args": params.endpointArgs,
-                    "locale": params.changeLang || displayOptions.language,
+                    "locale": params.changeLang || formplayerSessionPreferredLang || displayOptions.language,
```

Comment:
`displayOptions` is stored in local storage - if that already has the language why do we need a cookie?

Reply:
way better. I have reverted that and now updating `displayOptions` whenever the user changes the language in Webapps - [57dd4cf](https://github.com/dimagi/commcare-hq/pull/35444/commits/57dd4cf9d1bcaa65175bdb75b4158eff5bfee8e5)